src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -219,6 +232,11 @@ public ClusterName getClusterName() {
         return this.clusterName;
     }
 
+    // Used for testing and logging to determine how this cluster state was send over the wire
+    boolean fromDiff() {
```

Comment:
I think we should bake the use in the name like `wasReadFromDiff` and put a docstring on it